pkg/utils/node/predicates.go

```diff
@@ -45,6 +45,9 @@ func IsPastTTL(node *v1.Node) bool {
 // IsUnderutilized returns if the node has 0 non-daemonset pods
 func IsUnderutilized(node *v1.Node, pods []*v1.Pod) bool {
 	for _, pod := range pods {
+		if scheduling.HasFailed(pod) {
```

Comment:
This reads misleadingly. scheduling has failed? Can we move this library. wdyt `podanalysis.HasFailed` `podanalysis.IsOwnedbyDaemonSet`. Maybe `podutils`? and just name the package pod, and alias it where its used?